MapboxCoreNavigation/RouteController.swift

```diff
@@ -146,6 +146,18 @@ public protocol RouteControllerDelegate: class {
      */
     @objc(routeController:shouldPreventReroutesWhenArrivingAtWaypoint:)
     optional func routeController(_ routeController: RouteController, shouldPreventReroutesWhenArrivingAt waypoint: Waypoint) -> Bool
+    
+    
+    /**
+     Called when the route controller will disable battery monitoring.
+     
+     Implementing this method will allow developers to change whether battery monitoring is disabled when `RouteController` is deinited.
+     
+     - parameter routeController: The route controller that will change the state of battery monitoring.
+     - returns: A bool representing the value you would like `UIDevice.isBatteryMonitoringEnabled` set to.
+     */
+    @objc(routeControllerWillDisableBatteryMonitoring:)
+    optional func routeControllerWillDisableBatteryMonitoring(_ routeController: RouteController) -> Bool
```

Comment:
The logic here might be confusing. Typically, a delegate method with "will" or "should" in the name returns a boolean to indicate whether to proceed with the described action.

So, a `false` response to `routeControllerWillDisableBatteryMonitoring()` would mean *don't* disable it. This might be more clear if we change "will" to "should", but in either case I think the logic should be the other way around from how it is currently implemented here.